File: v131016/TeeNew/Series3D_AddArray.pas
unit Series3D_AddArray;
{$I TeeDefs.inc}

interface

uses
  {$IFNDEF LINUX}
  Windows, Messages,
  {$ENDIF}
  SysUtils, Classes,
  {$IFDEF CLX}
  QGraphics, QControls, QForms, QDialogs, QStdCtrls, QExtCtrls,
  {$ELSE}
  Graphics, Controls, Forms, Dialogs, StdCtrls, ExtCtrls,
  {$ENDIF}
  Base, TeEngine, TeeSurfa, TeeProcs, Chart;

type
  TAddArray3D = class(TBaseForm)
    Series1: TColorGridSeries;
    procedure FormCreate(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

implementation

{$IFNDEF CLX}
{$R *.dfm}
{$ELSE}
{$R *.xfm}
{$ENDIF}

procedure TAddArray3D.FormCreate(Sender: TObject);
var A : TArrayGrid;
    x,z : Integer;
begin
  inherited;

  // initialize dynamic array grid
  SetLength( A, 5, 7 ) ;

  // fill with random values
  for x:=0 to 4 do
      for z:= 0 to 6 do
          A[x,z] := Random ;

  // add to 3D series:
  Series1.AddArray( A );
end;

initialization
  RegisterClass(TAddArray3D);
end.

File: TeeNew/Builder2006/Series3D_AddArray.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#include <math.h>
#pragma hdrstop

#include "Series3D_AddArray.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "Base"
#pragma link "TeeSurfa"
#pragma resource "*.dfm"
TAddArray3D *AddArray3D;
//---------------------------------------------------------------------------
__fastcall TAddArray3D::TAddArray3D(TComponent* Owner)
  : TBaseForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TAddArray3D::FormCreate(TObject *Sender)
{
  TArrayGrid A;
  // initialize dynamic array grid
  A.Length =5;
  for (int t=0; t<A.Length-1;t++) A[t].Length = 7;

  // fill with random values
  for (int x=0;x<4;x++)
    for (int z=0;z<6;z++)
      A[x][z] = sqrt(pow(x,2)+pow(z-3,2)) ;

  // add to 3D series:
  Series1->AddArray(A);
}
//---------------------------------------------------------------------------

